src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -184,7 +184,8 @@ function validatePropTypes(element) {
 var ReactElementValidator = {
 
   createElement: function(type, props, children) {
-    var validType = typeof type === 'string' || typeof type === 'function';
+    var validType = typeof type === 'string' || typeof type === 'function' ||
+                    (type !== null && typeof type === 'object');
```

Comment:
This is going to be throwing later on if you tried to render this anywhere before. It is also going to be throwing through out version 15 so even if you try a later version and then downgrades, it'll still not warn. So I think this is pretty safe.

We can be more specific once we know a bit more about the data structures that will be allowed here. I suspect "module components" might end up here if we do those. As well as yields.